### PR TITLE
DM-16297: Changes for Gen3 support of LSST data

### PR DESCRIPTION
**Pull request by timj**

### CODE REVIEW

**Comment by timj** [comment]
I'm not really sure what to do about the changes to ingest.py. On the one hand it's clearly a bug in our headers that `INHERIT=T` is not set. On the other hand without this fix obs_lsst breaks. We don't want obs_lsst to have a special gen3 ingest task so what's the alternative?

**Comment by timj** [comment]
@czwa, @parejkoj this PR might be the most controversial bit of DM-16297. The problem is the forced merging of the primary header in ingest.py.  I don't think we want to have special ingest code for gen3 instruments so this merging (which uses override mode so should be safe) is the general answer in the case where data has forgotten to set INHERIT=T (which is the case for some LSST data).

**Comment by TallJimbo** [comment]
I don't have much of a problem with special code for different Gen3 instruments, though it's obviously best minimized.  The DM-17023 version of ingest already requires you to provide the fully-qualified `Instrument` class name via config parameter, so we have a place to hang such hooks (the registration system we had before was always a lie, as you had to import the module that defines that `Instrument` manually to make it work).

All that said, could we use `fix_headers` to add `INHERIT=T` instead?

**Comment by timj** [comment]
`fix_headers` runs after `afwFits.readMetadata` so that doesn't help. The code here to merge the headers doesn't actually harm anything since if INHERIT=T the merging would not change anything.  The change is if INHERIT=F and that was deliberate.

If ingestion already has to know the `Instrument` class it is ingesting then I could add a new property to `Instrument` with a boolean indicating whether merging should happen or not (note that `translate_header.py` always merges because astropy.io.fits does not automatically follow INHERIT=T whereas afw does).

**Comment by TallJimbo** [comment]
To clarify, the only danger here is that some other camera's data _intentionally_ has `INHERIT` missing (or `INHERIT=F`?), and this situation is that we just don't think that this is likely?

I don't think I'm particularly worried about that case, especially if we do the merging such that the extension header cards win when there are conflicts with the primary.  I guess I don't really have a problem with either approach.

**Comment by timj** [comment]
Yes. The second header always overwrites the first header so what my change does is mean that we always merge even if INHERIT=F and that is only going to be a problem if the first header is completely unrelated to the first empty header.  I'm okay with merging the headers now until we get to a point where an instrument turns up where that is bad.

File: python/lsst/obs/base/fitsRawFormatterBase.py [hunk 1]
```diff
@@ -163,6 +163,10 @@ def makeWcs(self, visitInfo, detector):
             Raised if there is an error generating the SkyWcs, chained from the
             lower-level exception if available.
         """
+        if self.observationInfo.tracking_radec is None:
+            # This is not an on-sky observation
+            return None
+
```
Comment (TallJimbo) [inline python/lsst/obs/base/fitsRawFormatterBase.py]: Might be a good idea to put this behind a function or `ObservationInfo` method, so if we need to expand it to e.g. `tracking_radec is None or <observation type is SCIENCE>` in order to make it more robust, we only have to do it in one place.

**Pull request merged**